### DISCUSSION TITLE
How to use class-validator for ObjectType (output data)?

### DISCUSSION THREAD

**Discussion by alex-kovalchuk** [MichalLytek/type-graphql#1173] [Q&A]
Hello. I need to validate input and output data.

```typescript
import { Field, InputType, ObjectType } from 'type-graphql';
import { IsString, IsUUID, Length, Matches } from 'class-validator';
import { EMAIL_PATTERN, UserLogin, UserLoginInput } from '../schemas/user.schema';

@InputType('UserLoginInput')
export class UserLoginInputDto implements UserLoginInput {
  @IsString()
  @Matches(EMAIL_PATTERN)
  @Length(6, 96)
  @Field({ description: `MinLen: 6, MaxLen: 96, Pattern: ${EMAIL_PATTERN.toString()}` })
  email: string;

  @IsString()
  @Length(8, 20)
  @Field({ description: 'MinLen: 8, MaxLen: 20' })
  password: string;
}

@ObjectType('UserLogin')
export class UserLoginDto implements UserLogin {
  @IsUUID('4')
  @Field({ description: 'UUID' })
  token: string;
}
```

If I pass an invalid UUID value to `UserLoginDto.token`, then this value will be returned without problems, but I expect a `GRAPHQL_VALIDATION_FAILED` or `INTERNAL_SERVER_ERROR` error. Then I can log this error and fix it.

Thanks.

**Comment by MichalLytek** [comment]
So you want to allow storying/calculating invalid data in all over your code but only return error that the used data is invalid?

I can't imagine a use case for output types validation. You should just make sure the values you create in your business code are correct, then just return.

**Reply by alex-kovalchuk** [reply]
This will be like a double-check. It is extremely important for me to ensure that the format of the outgoing data matches the description of the field.

Maybe there are some mechanisms for intercepting the result for adding a validator?

**Reply by MichalLytek** [reply]
You have middlewares that can intercept the result of the resolver or value of the field. You also have extensions with allows you to put some metadata for that field or query.

**Reply by alex-kovalchuk** [reply]
Thanks! I will try to do it with middleware.
Also thank you for type-graphql!
